src/shogun/mathematics/linalg/internal/implementation/Scale.h

```diff
@@ -66,6 +66,20 @@ struct scale
 	static void compute(Matrix A, Matrix B, Matrix C, T alpha, T beta);
 };
 
+/** Specialization of scale for the Native backend */
+template<> template<class Matrix>
+struct scale<Backend::NATIVE, Matrix>
+{
+		typedef typename Matrix::Scalar T;
+		
+		/** Performs the operation B = alpha*A */
+		static void compute(SGVector<T> A, SGVector<T> B, T alpha)
```

Comment:
This should be able to handle both SGMatrix and SGVector. We can have two methods
`static void compute(SGMatrix<T> A, SGMatrix<T> B, T alpha)`
and
`static void compute(SGVector<T> A, SGVector<T> B, T alpha)`
both of these would call internally a
`static void compute(T* A, T* B, T alpha)`
where you pass the number of elements you want to scale. That would be consistent with the Eigen3 and ViennaCL specializations we have for scale.